src/App.js

```diff
@@ -1,16 +1,35 @@
 import React from 'react';
-import './App.css';
+import { useState } from 'react';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
+import './App.css';
 
 const App = () => {
+  const [entries, setEntries] = useState(chatMessages);
+
+  const updateLikedEntry = (id) => {
+    setEntries((entries) =>
+      entries.map((entry) => {
+        if (entry.id === id) {
+          return { ...entry, liked: !entry.liked };
+        } else {
+          return entry;
+        }
+      })
+    );
+  };
+
+  const likes = entries.filter((entry) => entry.liked === true);
+  const likesCount = likes.length;
```

Comment:
Consider any variables that will change dynamically, turn them into state. React listens for state hooks, but not regular variables. Sometimes this can cause issues.
```python
const [likesCount, setLikesCount] = useState(0);
```
Refactor `likes` to be a function that will filter and then set state with `setLikesCount`